bricks/next-builder/src/data-providers/chunks/doLintStoryboard.ts

```diff
@@ -671,6 +735,116 @@ export function doLintStoryboard(storyboard: Storyboard): StoryboardError[] {
     });
   }
 
+  if (usingOldTemplate.size > 0) {
+    errors.push({
+      type: "warn",
+      code: "USING_OLD_TEMPLATE",
+      message: {
+        zh: "您正在使用旧的模板构件, 这种类型的构件在 v3 将决定废除，建议使用 Widget 或 自定义模板 构件进行代替: ",
+        en: "You are using old template components, which will be decided to be abolished in v3, Please use Widget or CustomTemplate instead: ",
+      },
+      list: [...usingOldTemplate.keys()],
+      details: [...usingOldTemplate.entries()].map(([message, meta]) => ({
+        message,
+        meta,
+      })),
+    });
+  }
+
+  if (usingProviderInRoute.size > 0) {
+    errors.push({
+      type: "warn",
+      code: "USING_PROVIDER_IN_ROUTE",
+      message: {
+        zh: "您在路由上面定义了 provider, 建议在 context 中使用 useProvider 替代:",
+        en: "You have defined Provider on the route, and this feature will be dropped in v3, Please use useProvider in route instead: ",
+      },
+      list: [...usingProviderInRoute.keys()],
+      details: [...usingProviderInRoute.entries()].map(([message, meta]) => ({
+        message,
+        meta,
+      })),
+    });
+  }
+
+  if (usingDefineResolve.size > 0) {
+    errors.push({
+      type: "warn",
+      code: "USING_DEFINE_RESOLVES_IN_ROUTE",
+      message: {
+        zh: "您在路由上面定义了 defineResolves, 建议在 context 中使用 useProvider 替代: ",
+        en: "You have defined defineResolves on the route, and this feature will be dropped in v3, Please use useProvider in route instead: ",
+      },
+      list: [...usingDefineResolve.keys()],
+      details: [...usingDefineResolve.entries()].map(([message, meta]) => ({
+        message,
+        meta,
+      })),
+    });
+  }
+  if (usingRedirectInRoute.size > 0) {
+    errors.push({
+      type: "warn",
+      code: "USING_DIRECT_IN_ROUTE",
+      message: {
+        zh: "您在路由上面定义了 redirect, 但路由类型非 redirect, 这将导致 redirect 不生效: ",
+        en: "You have defined redirect on the route, but the route type is not redirect, which will cause redirect to not take effect: ",
+      },
+      list: [...usingDefineResolve.keys()],
+      details: [...usingDefineResolve.entries()].map(([message, meta]) => ({
+        message,
+        meta,
+      })),
+    });
+  }
+
+  if (usingContextInBrick.size > 0) {
+    errors.push({
+      type: "warn",
+      code: "USING_CONTEXT_IN_BRICK",
+      message: {
+        zh: "您在构件上定义了 context, 建议将其迁移至路由上: ",
+        en: "You have defined the context on the component, Please migrate it to the route: ",
+      },
+      list: [...usingContextInBrick.keys()],
+      details: [...usingContextInBrick.entries()].map(([message, meta]) => ({
+        message,
+        meta,
+      })),
+    });
+  }
+
+  if (usingExportInBrick.size > 0) {
+    errors.push({
+      type: "warn",
+      code: "USING_EXPORTS_IN_BRICK",
+      message: {
+        zh: "您在构件上使用了 exports,该字段即将废弃, 建议使用 context 并指定 property 来声明一个上下文变量绑定到构件属性: ",
```

Comment:
context 指定 property 在 v3 中不支持

```suggestion
        zh: "您在构件上使用了已废弃的 exports 字段，该特性将在 v3 移除，建议使用 context 或 state 替代：",
```